### PR TITLE
Added javadoc generation to build file

### PR DESCRIPTION
**Pull request by magnus2lin**
@maxme as discussed, added javadoc generation to the build file. Building of the javadoc fails with 1 error because I have excluded `R.java`, where `SwipeToRefreshHelper.java` imports it. 

Furthermore, I'd like to add Android reference doc, just like the Java SE 7 document is added, but it has to be added locally in order for references to work.

### CODE REVIEW

**Comment by maxme** [comment]
I'm seeing 8 errors when trying to run `./gradlew generateReleaseJavadoc` in `WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java`

**Comment by magnus2lin** [comment]
Sorry, you are right. The errors occur because there is html code in some of the comments, (found in `HtmlUtils.java`) which I want to fix as well. Escaping the HTML makes the comments unreadable (when working with the class directly) whereas it looks good in the JavaDoc.  So I'm unsure which is the best way to move forward.

**Comment by magnus2lin** [comment]
All should be fine now. I will try to make things better next time, this is my first PR. I tried to edit comments containing HTML that makes both class's & javadoc readable.

File: WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java [hunk 1]
```diff
@@ -114,7 +125,11 @@ public static String stripScript(final String text) {
     }
 
     /**
-     * an alternative to Html.fromHtml() supporting <ul>, <ol>, <blockquote> tags and replacing EmoticonsUtils with Emojis
+     * An alternative to Html.fromHtml() supporting "ul" (&lt;ul&gt;), "ol" (&lt;ol&gt;), "blockquote" (&lt;blockquote&gt;)
```
Comment (maxme) [inline WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java]: We should use the `(@code)` notation here and in other places where html tags are used in comments, it makes things readable in the comment and in the generated html. For instance here: `{@code <ul>}, {@code <ol>},  {@code <ol>},  {@code <blockquote>},`

**Comment by magnus2lin** [comment]
Thanks!, @code blocks make things look much better.

File: WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java (hunk 1 above)
Comment (magnus2lin) [inline WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java]: Thanks!, @code blocks make things look much better. Updated the comments.

**Comment by maxme** [comment]
Thanks for the contrib!

:shipit:

**Pull request merged**